src/cli.ts

```diff
@@ -39,8 +43,13 @@ const main = async () => {
     },
   };
 
+  if (program.opts().privateKey) {
```

Comment:
It would be nice if the logs indicated which specific request received a 401 (e.g., was this on the initially discovery request that doesn't use the key at all?). If an incorrect key is used I'd expect to see a similar error. I think we can add that as a separate ticket though.

Initially, I was thinking it makes sense to allow the existing behavior because this is a prototype / testing tool and this could allow us to test unusual behavior, but testing against `https://bulk-data.smarthealthit.org/?err=token_invalid_token` I think the auth request is just skipped when the key isn't present.

I'll add a check that ensures that token URL, client ID, and private key are all provided if one is.

Reply:
An error is now thrown unless all or none of the auth options are provided:

```shell
npm run cli -- -f https://inferno.healthit.gov/reference-server/r4 -g 1a --token-url https://inferno.healthit.gov/reference-server/oauth/bulk-token

> bulk-data-export-client@0.0.1 cli
> ts-node --files src/cli.ts "-f" "https://inferno.healthit.gov/reference-server/r4" "-g" "1a" "--token-url" "https://inferno.healthit.gov/reference-server/oauth/bulk-token"

/Users/radamson/code/bulk-dqm/bulk-data-export-client/src/cli.ts:42
      throw new Error(
            ^
Error: Token URL, Client ID, and Private Key must all be provided or all omitted. Missing Client ID, Private Key
    at validateInputs (/Users/radamson/code/bulk-dqm/bulk-data-export-client/src/cli.ts:42:13)
    at main (/Users/radamson/code/bulk-dqm/bulk-data-export-client/src/cli.ts:61:3)
    at Object.<anonymous> (/Users/radamson/code/bulk-dqm/bulk-data-export-client/src/cli.ts:96:1)
    at Module._compile (node:internal/modules/cjs/loader:1099:14)
    at Module.m._compile (/Users/radamson/code/bulk-dqm/bulk-data-export-client/node_modules/ts-node/src/index.ts:1618:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/radamson/code/bulk-dqm/bulk-data-export-client/node_modules/ts-node/src/index.ts:1621:12)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
```